test/hotspot/jtreg/vmTestbase/nsk/jvmti/GetClassFields/getclfld007/getclfld007.cpp

```diff
@@ -125,61 +77,61 @@ jint Agent_Initialize(JavaVM *jvm, char *options, void *reserved) {
+    char *name, *sig;
     int j;
 
     if (jvmti == NULL) {
         printf("JVMTI client was not properly loaded!\n");
+        fflush(0);
         result = STATUS_FAILED;
         return;
     }
 
-    if (printdump == JNI_TRUE) {
-        printf(">>> %s:\n", classes[i].name);
-    }
+    // fieldArr contains 2 elements for each field
+    jint field_count = env->GetArrayLength(fieldArr) / 2;
 
     err = jvmti->GetClassFields(clazz, &fcount, &fields);
     if (err != JVMTI_ERROR_NONE) {
-        printf("(GetClassFields#%d) unexpected error: %s (%d)\n",
-               i, TranslateError(err), err);
+        printf("GetClassFields unexpected error: %s (%d)\n",
+               TranslateError(err), err);
+        fflush(0);
         result = STATUS_FAILED;
         return;
     }
 
-    if (fcount != classes[i].fcount) {
-        printf("(%d) wrong number of fields: %d, expected: %d\n",
-               i, fcount, classes[i].fcount);
+    if (fcount != field_count) {
+        printf("wrong number of fields: %d, expected: %d\n",
+               fcount, field_count);
         result = STATUS_FAILED;
     }
     for (j = 0; j < fcount; j++) {
         if (fields[j] == NULL) {
-            printf("(%d:%d) fieldID = null\n", i, j);
+            printf("(%d) fieldID = null\n", j);
+            result = STATUS_FAILED;
         } else {
-            err = jvmti->GetFieldName(clazz, fields[j],
-                &name, &sig, &generic);
+            err = jvmti->GetFieldName(clazz, fields[j], &name, &sig, NULL);
             if (err != JVMTI_ERROR_NONE) {
-                printf("(GetFieldName#%d:%d) unexpected error: %s (%d)\n",
-                       i, j, TranslateError(err), err);
+                printf("(GetFieldName#%d) unexpected error: %s (%d)\n",
+                       j, TranslateError(err), err);
             } else {
-                if (printdump == JNI_TRUE) {
-                    printf(">>>   [%d]: %s, sig = \"%s\"\n", j, name, sig);
-                }
-                if ((j < classes[i].fcount) &&
+                printf(">>>   [%d]: %s, sig = \"%s\"\n", j, name, sig);
+                if ((j < field_count) &&
                        (name == NULL || sig == NULL ||
-                        strcmp(name, classes[i].flds[j].name) != 0 ||
-                        strcmp(sig, classes[i].flds[j].sig) != 0)) {
-                    printf("(%d:%d) wrong field: \"%s%s\"", i, j, name, sig);
-                    printf(", expected: \"%s%s\"\n",
-                           classes[i].flds[j].name, classes[i].flds[j].sig);
+                        !equals_str(env, name, fieldArr, j * 2) ||
+                        !equals_str(env, sig, fieldArr, j * 2 + 1))) {
+                    printf("(%d) wrong field: \"%s%s\"", j, name, sig);
                     result = STATUS_FAILED;
                 }
+                jvmti->Deallocate((unsigned char *)name);
+                jvmti->Deallocate((unsigned char *)sig);
             }
         }
     }
```

Comment:
Nit: I'd suggested to refactor a little bit the content of loop at L111 as below:
```
    for (j = 0; j < fcount; j++) {
        if (fields[j] == NULL) {
            printf("(%d) fieldID = null\n", j);
            result = STATUS_FAILED;
            continue;                                                   => /you can consider to return here
        }
        err = jvmti->GetFieldName(clazz, fields[j], &name, &sig, NULL);
        if (err != JVMTI_ERROR_NONE) {
            printf("(GetFieldName#%d) unexpected error: %s (%d)\n",
                   j, TranslateError(err), err);
            result = STATUS_FAILED;                          => This result set is missed
            continue;                                                     => you can consider to return here
        }
        printf(">>>   [%d]: %s, sig = \"%s\"\n", j, name, sig);
        if ((j < field_count) &&
            (name == NULL || sig == NULL ||
             !equals_str(env, name, fieldArr, j * 2) ||
             !equals_str(env, sig, fieldArr, j * 2 + 1))) {
            printf("(%d) wrong field: \"%s%s\"", j, name, sig);
            result = STATUS_FAILED;
        }
        jvmti->Deallocate((unsigned char *)name);
        jvmti->Deallocate((unsigned char *)sig);
     }
```